File: 762 - Prime Number of Set Bits in Binary Representation.js
/**
 * @param {number} L
 * @param {number} R
 * @return {number}
 */
var countPrimeSetBits = function (L, R) {
    let res = 0;
    for (let i = L; i <= R; i++) {
        let bin = parseInt(i).toString(2);
        let count = 0;
        for (let bit of bin) {
            if (bit == '1') count++;
        }
        if (isprime(count)) res++;
    }
    return res;
};

function isprime(n) {
    if (n <= 1) return false;
    for (let i = 2; i < Math.floor(Math.sqrt(n)) + 1; i++) {
        if (n % i == 0) return false;
    }
    return true;
}


for (let i = 0; i < 20; i++) {
    let a = console.log(i + ": " + isprime(i));
}

console.log(parseInt(23).toString(2));

File: 762 - Prime Number of Set Bits in Binary Representation.py
import math


class Solution:
    def countPrimeSetBits(self, L: int, R: int) -> int:
        res = 0
        for i in range(L, R + 1):
            s = bin(i)
            count = 0
            for bit in s[2:]:
                if bit == '1':
                    count += 1
            if self.isprime(count):
                res += 1
        return res

    def isprime(self, n):
        if n <= 1:
            return False
        for i in range(2, int(math.sqrt(n)) + 1):
            if n % i == 0:
                return False
        return True


s = Solution()
for i in range(20):
    a = s.isprime(i)
    print(i, a)
